test/integration/version_upgrade_test.go

```diff
@@ -136,3 +140,83 @@ func TestVersionUpgrade(t *testing.T) {
 		t.Errorf("%s failed: %v", rr.Args, err)
 	}
 }
+
+// TestVersionUpgradeV1
+// downloads the minikube v1.0.0 binary and starts it
+// stops it
+// and then starts with head minikube binary
+func TestVersionUpgradeV1(t *testing.T) {
+	MaybeParallel(t)
+	profile := UniqueProfileName("vupgrade-v1")
+	ctx, cancel := context.WithTimeout(context.Background(), Minutes(55))
+
+	defer CleanupWithLogs(t, profile, cancel)
+
+	tf, err := ioutil.TempFile("", "minikube-v1-release.*.exe")
+	if err != nil {
+		t.Fatalf("tempfile: %v", err)
+	}
+	defer os.Remove(tf.Name())
+	tf.Close()
+
+	url := pkgutil.GetBinaryDownloadURL("v1.0.0", runtime.GOOS)
+	if err := retry.Expo(func() error { return getter.GetFile(tf.Name(), url) }, 3*time.Second, Minutes(3)); err != nil {
+		t.Fatalf("get failed: %v", err)
+	}
+
+	if runtime.GOOS != "windows" {
+		if err := os.Chmod(tf.Name(), 0700); err != nil {
+			t.Errorf("chmod: %v", err)
+		}
+	}
+
+	args := append([]string{"start", "-p", profile}, StartArgsLegacy()...)
+	rr := &RunResult{}
+	r := func() error {
+		rr, err = Run(t, exec.CommandContext(ctx, tf.Name(), args...))
+		return err
+	}
+
+	// Retry to allow flakiness for the previous release
+	if err := retry.Expo(r, 13*time.Second, Minutes(30), 2); err != nil {
+		t.Fatalf("v1.0.0 start failed: %v", err)
+	}
+	rr, err = Run(t, exec.CommandContext(ctx, tf.Name(), "stop", "-p", profile))
+	if err != nil {
+		t.Fatalf("%s failed: %v", rr.Args, err)
+	}
+
+	rr, err = Run(t, exec.CommandContext(ctx, tf.Name(), "-p", profile, "status", "--format={{.Host}}"))
+	if err != nil {
+		t.Logf("status error: %v (may be ok)", err)
+	}
+	got := strings.TrimSpace(rr.Stdout.String())
+	if got != state.Stopped.String() {
+		t.Errorf("status = %q; want = %q", got, state.Stopped.String())
+	}
+
+	args = append([]string{"start", "-p", profile, "--alsologtostderr", "-v=1"}, StartArgsLegacy()...)
+	rr, err = Run(t, exec.CommandContext(ctx, Target(), args...))
+	if err != nil {
+		t.Errorf("%s failed: %v", rr.Args, err)
+	}
+
+	s, err := Run(t, exec.CommandContext(ctx, "kubectl", "--context", profile, "version", "--output=json"))
+	if err != nil {
+		t.Fatalf("error running kubectl: %v", err)
+	}
+	cv := struct {
+		ServerVersion struct {
+			GitVersion string `json:"gitVersion"`
+		} `json:"serverVersion"`
+	}{}
+	err = json.Unmarshal(s.Stdout.Bytes(), &cv)
+	if err != nil {
+		t.Fatalf("error traversing json output: %v", err)
+	}
+	if _, err = semver.Parse(cv.ServerVersion.GitVersion); err != nil {
+		t.Fatalf("err parsing the kubernets version %v", err)
```

Comment:
Your going to need to strip the leading v:

`version_upgrade_test.go:218: err parsing the kubernets version Invalid character(s) found in major number "v1"`